connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/GrpcExceptionConverter.scala

```diff
@@ -47,11 +52,37 @@ private[client] object GrpcExceptionConverter {
     }
   }
 
-  private def toSparkThrowable(ex: StatusRuntimeException): SparkThrowable with Throwable = {
+  private val errorFactory = new ErrorFactoryBuilder()
+    .addConstructor((message, _) => new ParseException(None, message, Origin(), Origin()))
+    .addConstructor((message, _) => new AnalysisException(message))
+    .build()
+
+  private def errorInfoToThrowable(info: ErrorInfo, message: String): Throwable = {
+    val classes =
+      mapper.readValue(info.getMetadataOrDefault("classes", "[]"), classOf[Array[String]])
+
+    for (cls <- classes) {
+      errorFactory.get(cls) match {
+        case Some(constructor) =>
+          return constructor(message, null)
+        case None =>
+        // Do nothing
```

Comment:
Will we ever hit this?

Reply:
oh so this happens when there is nothing found from errorFactory so do nothing here means do not return then it fall back to `new SparkException(message)`

Reply:
But per design it will return on the first found error factory. Why it must be an array but just a string?

Reply:
Could you elaborate a little bit on the question?

Reply:
`val classes = mapper.readValue(info.getMetadataOrDefault("classes", "[]"), classOf[Array[String]])` 

this is a array of string. Why it is designed as array but just a single value?

Reply:
Because the classes included both the class name and all its parental class names of an exception.  It is possible that the exact match does not happen.

Reply:
hmmm I see. Sure should work for now.